hpat/datatypes/hpat_pandas_series_functions.py

```diff
@@ -2705,13 +2705,28 @@ def hpat_pandas_series_argsort(self, axis=0, kind='quicksort', order=None):
     if not (isinstance(axis, types.Omitted) or isinstance(axis, types.Integer) or axis == 0):
         raise TypingError('{} Unsupported parameters. Given axis: {}'.format(_func_name, axis))
 
+    if not isinstance(kind, (types.Omitted, str, types.UnicodeType, types.StringLiteral)):
+        raise TypingError('{} Non-string type unsupported. Given kind: {}'.format(_func_name, kind))
+
+    if not isinstance(order, (str, types.UnicodeType, types.StringLiteral, types.Omitted, types.NoneType, types.List))\
+            and order is not None:
+        raise TypingError('{} Unsupported parameters. Given order: {}'.format(_func_name, order))
+
     if not isinstance(self.index, types.NoneType):
         def hpat_pandas_series_argsort_idx_impl(self, axis=0, kind='quicksort', order=None):
-            sort = numpy.argsort(self._data, kind='mergesort')
+            if kind != 'quicksort' and kind != 'mergesort':
+                raise ValueError("Method argsort(). Unsupported parameter. Given 'kind' != 'quicksort' or 'mergesort'")
+            if kind == 'mergesort':
+                sort = numpy.argsort(self._data, kind='mergesort')
+            else:
+                sort = numpy.argsort(self._data)
```

Comment:
I think *if-statement* is redundant here, just use:
`sort = numpy.argsort(self._data, kind=kind)`

Reply:
We cant use kind=kind because numba gives typing error, I add a comment about it in code

Reply:
What typing error Numba provides? It is quite strange because kind='quicksort' by default

Reply:
Error: "numba.errors.TypingError: Failed in nopython mode pipeline (step: nopython frontend)                                                    Internal error at <numba.typeinfer.CallConstraint object at 0x000001878B10F908>.                                                        Failed in nopython mode pipeline (step: nopython mode backend)                                                                          No definition for lowering <function argsort at 0x0000018721BE6438>(array(int64, 1d, C), unicode_type) -> array(int64, 1d, C)"

If I submit the variable kind, it is a Unicode type, not StringLiteral